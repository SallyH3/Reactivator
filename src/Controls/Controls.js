//this is where study list button will live

//will display message to user if they click the button and there are no incorrect cards to display