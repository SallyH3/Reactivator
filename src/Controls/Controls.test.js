import React from 'react';
import Controls from './Controls';
import { shallow } from 'enzyme';

const displayIncorrectCardsMock = jest.fn();

const mockIncorrectCards = [];

describe('Controls', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Controls 
      incorrectCards={mockIncorrectCards}
      displayIncorrectCards={displayIncorrectCardsMock}/>
    )
  });

  it ('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('renders the study list button', () => {
    let btn = (wrapper).find('.study-button').simulate('click')
    expect(btn.exists()).toEqual(true);
  });
});