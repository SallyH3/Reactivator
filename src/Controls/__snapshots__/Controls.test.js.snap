// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should match snapshot 1`] = `
<section
  className="button-container"
>
  <input
    className="study-button"
    type="submit"
    value="Study list"
  />
  <input
    className="show-all-button"
    type="submit"
    value="Show all"
  />
</section>
`;
